library/include/rocwmma/internal/transforms_impl.hpp

```diff
@@ -181,9 +185,88 @@ namespace rocwmma
         lo          = Blend::Zip32::exec(lo, rot_hi);
         hi          = Blend::Zip32::exec(rot_lo, hi);
 
-        return PackUtil::template paddedUnpack<VecSize>(concat(lo, hi));
+        return concat(PackUtil::template paddedUnpack<VecSize / 2u>(lo),
+                      PackUtil::template paddedUnpack<VecSize / 2u>(hi));
     }
 
+    template <typename DataT, uint32_t VecSize>
+    ROCWMMA_DEVICE constexpr static inline auto aos_soa_wave_b32_vw4(VecT<DataT, VecSize> const& v)
+    {
+        using PackUtil = PackUtil<DataT>;
+
+        // Step 1 : UnpackLohi16
+        auto packed_data = PackUtil::paddedPack(concat(extractEven(v), extractOdd(v)));
+
+        packed_data = Swizzle::RotateR32<16>::exec(packed_data);
+
+        auto unpacked_data = PackUtil::template paddedUnpack<4>(packed_data);
```

Comment:
Can we use the UnpackLoHi16 function here?

Reply:
done